test/integration/test_controlplane.go

```diff
@@ -392,6 +397,186 @@ func TestControlPlaneEssentials(t *testing.T) {
 	)
 }
 
+func TestControlPlaneWatchNamespaces(t *testing.T) {
+	t.Parallel()
+	namespace, cleaner := helpers.SetupTestEnv(t, GetCtx(), GetEnv())
+
+	dataplaneClient := GetClients().OperatorClient.GatewayOperatorV1beta1().DataPlanes(namespace.Name)
+	controlplaneClient := GetClients().OperatorClient.GatewayOperatorV1beta1().ControlPlanes(namespace.Name)
+
+	dp := &operatorv1beta1.DataPlane{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace:    namespace.Name,
+			GenerateName: "dp-watchnamespaces-",
+		},
+		Spec: operatorv1beta1.DataPlaneSpec{
+			DataPlaneOptions: operatorv1beta1.DataPlaneOptions{
+				Deployment: operatorv1beta1.DataPlaneDeploymentOptions{
+					DeploymentOptions: operatorv1beta1.DeploymentOptions{
+						PodTemplateSpec: &corev1.PodTemplateSpec{
+							Spec: corev1.PodSpec{
+								Containers: []corev1.Container{
+									{
+										Name:  consts.DataPlaneProxyContainerName,
+										Image: helpers.GetDefaultDataPlaneImage(),
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+	t.Log("deploying dataplane resource")
+	dp, err := dataplaneClient.Create(GetCtx(), dp, metav1.CreateOptions{})
+	require.NoError(t, err)
+	cleaner.Add(dp)
+
+	createNamespace := func(t *testing.T, cl client.Client, cleaner *clusters.Cleaner, generateName string) *corev1.Namespace {
+		ns := &corev1.Namespace{
+			ObjectMeta: metav1.ObjectMeta{
+				GenerateName: generateName,
+			},
+		}
+		require.NoError(t, cl.Create(GetCtx(), ns))
+		cleaner.AddNamespace(ns)
+		return ns
+	}
+	nsA := createNamespace(t, GetClients().MgrClient, cleaner, "test-namespace-a")
+	nsB := createNamespace(t, GetClients().MgrClient, cleaner, "test-namespace-b")
+
+	cp := &operatorv1beta1.ControlPlane{
```

Comment:
Not saying we need to do it right now, but it would be cool to have a builder for the CP as well